gitlab/utils.py

```diff
@@ -100,33 +110,40 @@ def __init__(
         self.retry_transient_errors = retry_transient_errors
 
     def _retryable_status_code(
-        self,
-        result: requests_backend.RequestsResponse,
+        self, status_code: Optional[int], reason: str = ""
```

Comment:
I dumbed this down from a `Response` object to individual arguments so we can feed it in from different kinds of clients more easily.

Coincidentally that simplifes the unit tests for retries quite a bit.